### PR TITLE
txt-preview: improvments

### PR DESCRIPTION
**Pull request by dfdan**
:heart: Thank you for your contribution!

### Description

* Fixes horizontal scrolling in preview iframe
* Adds option to truncate .txt preview after a specified number of bytes

Fixes invenio-app-rdm#1864

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Review by tmorrell — APPROVED** [review]
Awesome! I've rolled this out to out production site already, since it provides a much improved experience. Example: https://data.caltech.edu/records/1g67b-34e79

File: invenio_previewer/extensions/txt.py
Comment (fenekku) [inline invenio_previewer/extensions/txt.py:24]:
```suggestion
        return fp.read(max_bytes).decode(encoding, errors="ignore")
```

Selecting an arbitrary number of bytes could truncate content in the middle of a multi-byte character and produce errors when decoding (e.g. `b'\xe2\x99\x9c\n'` -"♜\n"- could be cut at `b'\xe2\x99` and that won't decode properly). The easiest is to ignore decoding errors so that this case doesn't raise and invalid decodings are just ignored (no "�" glyph). It may mask other decoding issues, but that just makes the previewer more resilient.

**Review by fenekku — APPROVED** [review]
Thanks for this! Approved pending comments.

File: invenio_previewer/extensions/txt.py
Comment (dfdan) [reply invenio_previewer/extensions/txt.py]: Thanks picked this / some extra spaces up in tests.

**Comment by dfdan** [comment]
Thanks @fenekku, good thinking on the possible multi-byte errors.
I've added some tests but it looks like test_macros.py fails all tests anyway :( Tom has demonstrated it works however. 

It would be good if we could merge this and @tmorrell's #179 into a 1.3.10 release (as well as push these changes forwards into 1.4) - so there is a painless upgrade option for existing installs.

**Comment by fenekku** [comment]
Looping in Karolina from CERN. Once this one and Tom's one are merged let do a release and backport to 1.3.10 like Dan mentioned.

File: invenio_previewer/templates/semantic-ui/invenio_previewer/txt.html
Comment (ntarocco) [inline invenio_previewer/templates/semantic-ui/invenio_previewer/txt.html:13]: For the future: maybe we can simply use the autogenerated classes? See [here](https://github.com/inveniosoftware/invenio-theme/blob/d44a5b72eeb0716ec8081731df552fafa8f8a3de/invenio_theme/assets/semantic-ui/less/invenio_theme/theme/globals/site.overrides#L358)
```suggestion
<div class="rel-p-1">
```

**Review by ntarocco — APPROVED** [review]

**Pull request merged**
